compiler/util/files.cpp

```diff
@@ -327,13 +328,18 @@ void closefile(fileinfo* thefile) {
 }
 
 
-void openCFile(fileinfo* fi, const char* name, const char* ext) {
+void openCFile(fileinfo* fi, const char* name, const char* ext,
+               bool useTmpDir) {
   if (ext)
     fi->filename = astr(name, ".", ext);
   else
     fi->filename = astr(name);
 
-  fi->pathname = genIntermediateFilename(fi->filename);
+  if (useTmpDir) {
+    fi->pathname = genIntermediateFilename(fi->filename);
+  } else {
+    fi->pathname = astr(name);
+  }
```

Comment:
I don't think you should change this function to make Makefiles - it's for making C files - see my other comment in the header